lib/OpenQA/Schema.pm

```diff
@@ -100,19 +87,12 @@ sub deploy {
 }
 
 # Class attribute used for testing with OpenQA::Test::Database
-sub search_path_for_tests {
-    my $class = shift;
-    state $search_path;
-    $search_path = shift if @_;
-    return $search_path;
-}
+sub search_path_for_tests ($class, $new_search_path = undef) { state $search_path = $new_search_path }
```

Comment:
This doesn't work the way you think it is.
Try:
```
$x = search_path_for_tests(x => "abc");
say $x;
$x = search_path_for_tests(x => "abcd");
say $x;
$x = search_path_for_tests("x");
say $x;'
```

```suggestion
sub search_path_for_tests ($class, $new_search_path = undef) {
  state $search_path;
  $search_path = $new_search_path if defined $new_search_path; 
  return $search_path;
}
```

Reply:
Good catch. 🎣

Reply:
done